src/http/endpoints/update_profile.rs

```diff
@@ -25,50 +25,63 @@ pub async fn update_profile(
         return Ok(HttpResponse::Unauthorized().finish());
     }
 
-    let instantiated_entity = ayb_db.get_entity_by_slug(entity_slug).await?;
-    let links = if let Some(web_info) = Option::as_ref(&**web_info) {
-        // If there's a known web frontend, we verify the identity of the links.
-        let mut links = vec![];
-        for link in profile.links.into_iter() {
-            let url = Url::parse(&link.url)?;
-            links.push(Link {
-                url: link.url,
-                verified: is_verified_url(
-                    url,
-                    Url::from_str(&web_info.profile(entity_slug))
-                        .expect("invalid web frontend url"),
-                )
-                .await,
-            })
-        }
+    let links = if let Some(profile_links) = profile.links {
+        if let Some(web_info) = Option::as_ref(&**web_info) {
+            // If there's a known web frontend, we verify the identity of the links.
+            let mut links = vec![];
+            for link in profile_links.into_iter() {
+                let url = Url::parse(&link.url)?;
+                links.push(Link {
+                    url: link.url,
+                    verified: is_verified_url(
+                        url,
+                        Url::from_str(&web_info.profile(entity_slug))
+                            .expect("invalid web frontend url"),
+                    )
+                    .await,
+                })
+            }
 
-        Some(links)
+            Some(links)
+        } else {
+            // If there is no known web frontend, we save links to the profile without verification.
+            Some(
+                profile_links
+                    .into_iter()
+                    .map(|l| Link {
+                        url: l.url,
+                        verified: false,
+                    })
+                    .collect(),
+            )
+        }
     } else {
-        // If there is no known web frontend, we save links to the profile without verification.
-        Some(
-            profile
-                .links
-                .into_iter()
-                .map(|l| Link {
-                    url: l.url,
-                    verified: false,
-                })
-                .collect(),
-        )
+        None
     };
 
-    let entity = Entity {
-        slug: instantiated_entity.slug,
-        entity_type: instantiated_entity.entity_type,
-        display_name: profile.display_name,
-        description: profile.description,
-        organization: profile.organization,
-        location: profile.location,
-        links,
-    };
+    let mut partial = PartialEntity::new();
+    if let Some(display_name) = profile.display_name {
+        partial.display_name = Some(Some(display_name));
```

Comment:
Two related questions:
* Why do we need `Some(Some(...))` here and below?
* Given the data coming off the wire contains `Option`s, can't we copy them from the struct being send from the client?

Reply:
The idea of `Option<Option<T>>` is the following:
* Value is `None` &rarr; do not change anything.
* Value is `Some(None)` &rarr; set the value to `NULL`.
* Value is `Some(v)` &rarr; set the value to `v`.

We can't directly use the data coming off the wire because JSON doesn't have a way to represent an `Option<Option<T>>` :)
I could simplify this implementation using `map`, I think I'm going to update it :)

Reply:
Great explanation, and smart encoding! I have a slight preference for excluding the key entirely to represent no update, and then you only need a single level of `Option`. That said, I'll leave a comment below mentioning to document the encoding if you keep it this way.

Reply:
I'm going to keep this struct as-is, because I don't really like the idea of throwing a `HashMap` around. But I've updated the implementation of the update entity endpoint and now HTTP clients can do exactly that:
* Do not include the key &rarr; `None`
* Key is `null` &rarr; `Some(None)`
* Key is `v` &rarr; Some(Some(v))

I'll add the docstring :)